### ISSUE TITLE
No scopes on periods

### ISSUE THREAD

**Issue by chbk** [MagicStack/MagicPython#98]
* **Editor name and version**: Atom 1.18.0
* **Platform**: macOS 10.12.6
* **Color scheme**: custom
* **MagicPython version**: 1.0.11
* **A sreenshot**: 
![p](https://user-images.githubusercontent.com/7478535/29130043-fdeb6f68-7d28-11e7-8d57-9c0e2e552699.png)

* **5-10 lines of surrounding code**:

```
foo(fizz.buzz)              # period is meta.function-call.arguments.python

instance.method()           # period is not scoped

'string {}'.format(param)   # period is not scoped
```

**Comment by 1st1** [comment]
Why do you want them scoped? Does the built-in `language-python` scope periods?

**Comment by chbk** [comment]
It doesn't and I consider that a bug too. Other language packages scope periods. Also, braces and brackets are scoped, so periods should be too imo. I personally want them scoped as my syntax theme colors punctuation in white.

**Comment by 1st1** [comment]
Yeah, makes sense. We'll look into this. /cc @vpetrovykh

**Comment by vpetrovykh** [comment]
The two of the more obvious choices for scope prefix are: `punctuation` (like the various brackets and parentheses) and `keyword.operator` (like the `+`). The purist in me wants to call it an operator, since that's what it actually is in Python, but the pragmatist suggests that calling `.` an operator may look terrible in many existing color schemes that highlight other operators (so I'd likely end up calling it something like `punctuation.accessor.python`).

If you have any thoughts or suggestions regarding the naming, please let me know here.

You can refer to scopes described here: https://manual.macromates.com/en/language_grammars. Historically, that's one of the descriptions of scopes that this grammar relied upon. If you know of a better document on scope name standards, please do provide a link.

**Comment by chbk** [comment]
Most packages scope them as `punctuation.separator` or `punctuation.delimiter`. The naming convention for Atom is [here](https://github.com/atom/flight-manual.atom.io/pull/114).

**Comment by vpetrovykh** [comment]
Thanks for the link. I can see how `punctuation.separator` may be kind of reasonable for `.`.

**Comment by 1st1** [comment]
`punctuation.separator` is fine by me.

**Comment by 1st1** [comment]
We'll need to look how this change will look like with stock sublime and atom themes before making a decision through.

**Comment by 1st1** [comment]
Please research what other syntaxes use for periods, and how GitHub highlights periods for those languages. We don't want all `.` to be suddenly highlighted in red or blue on GitHub. All in all, we need to be extra careful with our choices here, as it may affect a huge number of people.

**Comment by chbk** [comment]
[List of grammars](https://github.com/github/linguist/blob/master/vendor/README.md) used on GitHub.

List of languages sorted by popularity on GitHub:
- [Javascript](https://github.com/atom/language-javascript) scopes as [`meta.delimiter.method.period`](https://github.com/atom/language-javascript/blob/b01b9b6e445428438d259b336596058d30f28751/grammars/javascript.cson#L1197) and [`meta.delimiter.property.period`](https://github.com/atom/language-javascript/blob/b01b9b6e445428438d259b336596058d30f28751/grammars/javascript.cson#L1814)
    ```javascript
    foo.bar()
    fizz.buzz
    ```
    Same package used for Atom

- [Java](https://github.com/textmate/java.tmbundle) scopes as [`punctuation.separator`](https://github.com/textmate/java.tmbundle/blob/bd222183660baffd9fab21a0d8230be0e07322b8/Syntaxes/Java.plist#L89)
    ```java
    import fizz.buzz
    ```
    [language-java](https://github.com/atom/language-java) scopes as [`punctuation.separator.period`](https://github.com/atom/language-java/blob/652e8ba41a43d21fc92477da699fb99a655c7908/grammars/java.cson#L411) in Atom

- [Python](https://github.com/MagicStack/MagicPython) doesn't scope
    ```python
    foo.bar()
    fizz.buzz
    ```
    [language-python](https://github.com/atom/language-python) doesn't scope in Atom

- [Ruby](https://github.com/aroben/ruby.tmbundle) scopes as [`punctuation.separator.method`](https://github.com/aroben/ruby.tmbundle/blob/3d4213e5e0a437de5b75bf6d2ba625ca14e0c309/Syntaxes/Ruby.plist#L2506)
    ```ruby
    foo::bar
    fizz.buzz()
    ```
    [language-ruby](https://github.com/atom/language-ruby) scopes as [`punctuation.separator.method`](https://github.com/atom/language-ruby/blob/035fbc18a223bfac6ce9a3bf51d0cc65317fd37d/grammars/ruby.cson#L1946) in Atom

- [PHP](https://github.com/textmate/php.tmbundle) scopes as [`keyword.operator.string`](https://github.com/textmate/php.tmbundle/blob/4b5c247c34f6c73e193ec810f320acad8a61b263/Syntaxes/PHP.plist#L2284)
    ```php
    foo.bar()
    fizz.buzz
    ```
    [language-php](https://github.com/atom/language-php) scopes as [`keyword.operator.string`](https://github.com/atom/language-php/blob/a758cf2cc6b7fe2295abfa4d9676f32db026c676/grammars/php.cson#L1960) in Atom

- [C & C++](https://github.com/textmate/c.tmbundle) doesn't scope
    ```c
    {
        foo.bar()
        fizz.buzz
    }
    ```
    [language-c](https://github.com/atom/language-c) scopes as [`punctuation.separator.dot-access`](https://github.com/atom/language-c/blob/6038fa39738f6f9fe9bbf4988271c464edf75e97/grammars/c.cson#L309) in Atom

- [CSS](https://github.com/atom/language-css/) scopes as [`punctuation.definition.entity`](https://github.com/atom/language-css/blob/8d8c6c1a0a995521768c8330715f4d5c8d725c35/grammars/css.cson#L1793)
    ```css
    .class {}
    ```
    Same package used in Atom

- [C#](https://github.com/atom/language-csharp) doesn't scope
    ```c#
    {
        foo.bar()
        fizz.buzz
    }
    ```
    Same package used in Atom

**Comment by vpetrovykh** [comment]
OK, let's go with `punctuation.separator.period.python`.

**Comment by 1st1** [comment]
@chbk Thanks a lot for the thorough research! We'll likely go with `punctuation.separator.period`.

**Comment by chbk** [comment]
Awesome :smile:

**Comment by vpetrovykh** [comment]
Incidentally, do you think that the scope of `.` in imports should be the same as everywhere else?

```python
import foo.bar
from . import baz
from .spam import ham
```

**Comment by 1st1** [comment]
Yeah, why not.

**Comment by 1st1** [comment]
I've tested the updated grammar with all built-in Sublime Text themes, and built-in and top Atom themes. It seems that about 10% of colour schemas do highlight `punctuation.separator.period.python` scope differently from `source.python`, but none do it in a disturbing way.

**Comment by 1st1** [comment]
I've just released MagicPython 1.0.12 with this change in. Enjoy!
